rotkehlchen/chain/ethereum/balancer/balancer.py

```diff
@@ -0,0 +1,321 @@
+                    user_balance=user_asset_balance,
+                    asset=asset,
+                )
+                bpool_assets.append(bpool_asset)
+
+            bpool = BalancerPool(
+                address=to_checksum_address(pool['id']),  # ! Checksum address
+                assets=bpool_assets,
+                assets_count=FVal(pool['tokensCount']),
+                symbol=pool['symbol'],
+                balance=pool_balance,
+                weight=FVal(pool['totalWeight']),
+                user_balance=user_pool_balance,
+            )
+            addresses_balancer_pools[address].append(bpool)
+
+        return BalancerBalances(
+            addresses_balancer_pools=addresses_balancer_pools,
+            known_assets=known_assets,
+            unknown_assets=unknown_assets,
+        )
+
+    def _get_known_assets_prices(
+        self,
+        known_assets: Set[KnownAsset],
+        unknown_assets: Set[ChecksumEthAddress],
+    ) -> AssetsPrices:
+        """Get the tokens prices via Inquirer
+
+        Given an asset, if `find_usd_price()` returns zero, it will be added
+        into `unknown_assets`.
+        """
+        known_asset_prices: AssetsPrices = {}
+        known_assets_ = list(known_assets)  # ! Get a deterministic sequence
+        # ! TODO VN PR: consider timeout and list/kill greenlets
+        assets_usd_prices = [
+            gevent.spawn(Inquirer().find_usd_price, known_asset.asset)
+            for known_asset in known_assets_
+        ]
+        gevent.joinall(assets_usd_prices, timeout=5)
+
+        for known_asset, asset_usd_price in zip(known_assets_, assets_usd_prices):
+            if asset_usd_price.value != Price(ZERO):
+                known_asset_prices[known_asset.address] = FVal(asset_usd_price.value)
+            else:
+                unknown_assets.add(known_asset.address)
+
+        return known_asset_prices
+
+    def _get_unknown_assets_prices(
+            self,
+            unknown_assets: Set[ChecksumEthAddress],
+    ) -> AssetsPrices:
+        """Get the tokens prices via the Balancer subgraph
+
+        Query
+        -----
+        Get list of tokenPrices filtering by token id
+
+        # ! TODO VN PR
+        # ! Add USD value per asset, either via PriceHistorian or The Graph
+        # ! Add Asset instance per asset, at least if exists
+        """
+        unknown_assets_prices: AssetsPrices = {}
+        # ! Format addresses, The Graph does not support checksum addresses
+        token_addresses = [address.lower() for address in unknown_assets]
+        param_types = {
+            '$token_ids': '[ID!]',
+        }
+        param_values = {
+            'token_ids': token_addresses,
+        }
+        result = self.graph.query(
+            querystr=TOKENPRICES_QUERY.format(),  # ! Remove returns in string
+            param_types=param_types,
+            param_values=param_values,
+        )
+        # ! Checksum address
+        unknown_assets_prices = {
```

Comment:
are we sure the prices returned by the graph are in USD? Did you check that somewhere?

Reply:
Good one. I've just asked in Balancer discord as the [subgraph documentation](https://github.com/balancer-labs/balancer-subgraph/blob/master/schema.graphql#L109) is not clear. I'd say it is in USD, but better ask.

Reply:
I confirmed on discord that price is in USD